pennylane/qchem/vibrational/pes_generator.py

```diff
@@ -0,0 +1,676 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains functions to calculate potential energy surfaces
+per normal modes on a grid."""
+
+import itertools
+import subprocess
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.data.base._lazy_modules import h5py
+
+from .vibrational_class import get_dipole, single_point
+
+# pylint: disable=too-many-arguments, too-many-function-args, c-extension-no-member
+# pylint: disable= import-outside-toplevel, too-many-positional-arguments
+
+# constants
+HBAR = 6.022 * 1.055e12  # (amu)*(angstrom^2/s)
+C_LIGHT = 3 * 10**8  # m/s
+AU_TO_CM = 219475
+BOHR_TO_ANG = 0.5291772106
+
+
+def _import_mpi4py():
+    """Import mpi4py."""
+    try:
+        import mpi4py
+    except ImportError as Error:
+        raise ImportError(
+            "This feature requires mpi4py. It can be installed with: pip install mpi4py."
+        ) from Error
+
+    return mpi4py
+
+
+def pes_onemode(
+    molecule, scf_result, freqs_au, displ_vecs, gauss_grid, method="rhf", do_dipole=False
+):
+    r"""Computes the one-mode potential energy surface on a grid in real space, along the normal coordinate directions
+    (or any directions set by the displ_vecs).
+    Simultaneously, can compute the dipole one-mode elements.
+
+    Args:
+       molecule: Molecule object.
+       scf_result: pyscf object from electronic structure calculations
+       freqs_au: list of normal mode frequencies
+       displ_vecs: list of displacement vectors for each normal mode
+       gauss_grid: sample points for Gauss-Hermite quadrature
+       method: Electronic structure method to define the level of theory
+            for harmonic analysis. Default is restricted Hartree-Fock 'rhf'.
+       do_dipole: Whether to calculate the dipole elements. Default is ``False``.
+
+    Returns:
+      A tuple of one-mode potential energy surface and one-mode dipole along
+      the normal-mode coordinates.
+
+    """
+
+    _import_mpi4py()
+    from mpi4py import MPI
+
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+
+    local_pes_onebody, local_dipole_onebody = _local_pes_onemode(
+        comm,
+        molecule,
+        scf_result,
+        freqs_au,
+        displ_vecs,
+        gauss_grid,
+        method=method,
+        do_dipole=do_dipole,
+    )
+
+    filename = f"v1data_{rank}.hdf5"
```

Comment:
We should not be saving the data by default.

Reply:
This is not for saving purpose, this is saved for different threads so that we can later collect it using the load functions. This file in fact gets removed once its purpose is served.